### PR TITLE
Capture balance reason

### PR DESCRIPTION
**Pull request by maoueh**
This is commits I have locally that I would like to share with you prior the meeting

### CODE REVIEW

File: core/state/state_object.go [hunk 1]
```diff
@@ -54,6 +54,26 @@ func (s Storage) Copy() Storage {
 	return cpy
 }
 
+type BalanceChangeReason byte // Reason for the balance change
+
+const (
+	BalanceChangeUnspecified          BalanceChangeReason = 0x0
```
Comment (s1na) [inline core/state/state_object.go]:
```golang
const (
    BalanceChangeUnspecified BalanceChangeReason = iota
    BalanceChangeRewardMineUncle
    BalanceChangeRewardMineBlock
    ...
)
```

File: core/vm/evm.go [hunk 2]
```diff
@@ -175,6 +176,13 @@ func (evm *EVM) SetBlockContext(blockCtx BlockContext) {
 // the necessary steps to create accounts and reverses the state in case of an
 // execution error or failed value transfer.
 func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+	if v, ok := evm.Config.Tracer.(EVMLoggerExtended); evm.Config.Tracer != nil && ok {
```
Comment (s1na) [inline core/vm/evm.go]: Please leave out these for now.

**Comment by maoueh** [comment]
PR updated

File: core/vm/evm.go (hunk 2 above)
Comment (maoueh) [reply core/vm/evm.go]: FYI, I have locally the change that moves up existing `CaptureStart/Enter`, just let me know if you need it.

**Comment by s1na** [comment]
Thanks!

**Pull request merged**